### PR TITLE
Made go snake compatible with new api

### PR DESCRIPTION
**Pull request by zrdaley**
This PR does the following:
- makes the snake compatible with the new api as specified [here](https://github.com/battlesnakeio/docs/blob/master/apis/snake/spec.yaml)
- removes dead code
- simplifies the move handler so that the snake always default to down

Related issues: https://github.com/battlesnakeio/roadmap/issues/38

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -5,115 +5,46 @@ import (
...
-func (list *SnakeList) UnmarshalJSON(data []byte) error {
-	var obj struct {
-		Data []Snake `json:"data"`
-	}
-	if err := json.Unmarshal(data, &obj); err != nil {
-		return err
-	}
-	*list = obj.Data
-	return nil
+func DecodeSnakeRequest(req *http.Request) (*SnakeRequest, error) {
```
Comment (joram) [inline api.go]: idiomatic go is to call these types of function `New<ModeName>` I believe.

File: api_test.go [hunk 2]
```diff
@@ -16,233 +48,43 @@ func requestWithBody(body string) *http.Request {
...
-	`
-
-	req := requestWithBody(s)
-	result, err := NewMoveRequest(req)
+func TestDecodeSnakeRequest(t *testing.T) {
+	req := requestWithBody(mockReq)
+	result, err := DecodeSnakeRequest(req)
 
 	if err != nil {
 		t.Fatal(err)
```
Comment (joram) [inline api_test.go]: have you seen the `require` and `assert` libraries? I would suggest using `require.NotError(t, err)` instead here. It returns nice error messages. and is more instructive. :)

File: routes.go [hunk 3]
```diff
@@ -2,50 +2,33 @@ package main
...
-
-	r := rand.New(rand.NewSource(time.Now().UnixNano()))
-
 	respond(res, MoveResponse{
-		Move: directions[r.Intn(4)],
+		Move: "down",
 	})
 }
+
+func End(res http.ResponseWriter, req *http.Request) {
```
Comment (joram) [inline routes.go]: should we have a request to decode here?

Comment (zrdaley) [reply routes.go]: Honestly, I wasn't really sure what to put here. The api spec has the following info about the /end response: 
```      
responses:
        default:
          description: "Response ignored"
```

So shouldn't it just do nothing?

Comment (joram) [reply routes.go]: yeah, you are right, no action needs to be done. :+1:

**Pull request merged**